test/CMakeLists.txt

```diff
@@ -56,6 +56,41 @@ elseif (PORTS_OF_CALL_TEST_PORTABILITY_STRATEGY STREQUAL "Cuda")
   target_compile_definitions(portsofcall_iface INTERFACE PORTABILITY_STRATEGY_CUDA)
 endif()
 
+if(NOT PORTS_OF_CALL_SILENCE_WARNINGS)
+    # turn on strict warnings and error
+    # NOTE: The ROCm 6.2.1 installation on some machines (RZAdams and RZVernal at least) will
+    #       inject a `--gcc-toolchain` command-line argument into Clang, which is a deprecated
+    #       option.  Clang will warn about unused command-line arguments, so you get multiple
+    #       instances of that warning.  Then `-Werror` promotes those to errors.  We can turn these
+    #       back to non-fatal warnings with `-Wno-error=unused-command-line-argument` or we can
+    #       turn off that particular warning with `-Wno-unused-command-line-argument`.
+    # TODO: find comiler flags for non-GNU compatible compilers
+    target_compile_options(portsofcall_iface INTERFACE
+        $<$<IN_LIST:${CMAKE_CXX_COMPILER_ID},GNU>:-Werror;-Wall;-pedantic-errors;-Wunused-parameter>
+        $<$<IN_LIST:${CMAKE_CXX_COMPILER_ID},AppleClang>:-Werror;-Wall;-Wunused-parameter>
+        $<$<NOT:$<BOOL:${WIN32}>>:$<$<IN_LIST:${CMAKE_CXX_COMPILER_ID},Clang>:-Werror;-Wall;-Wno-unused-command-line-argument>>
+        $<$<BOOL:${WIN32}>:$<$<IN_LIST:${CMAKE_CXX_COMPILER_ID},Clang>:-Wall;-Wno-c++98-compat;-Wno-c++98-compat-pedantic>>
+        $<$<IN_LIST:${CMAKE_CXX_COMPILER_ID},Intel>:-Werror;-w2;-Wunused-variable;-Wunused-parameter;-diag-disable=remark>
+        $<$<IN_LIST:${CMAKE_CXX_COMPILER_ID},PGI>:>
+        $<$<IN_LIST:${CMAKE_CXX_COMPILER_ID},MSVC>:/permissive->
+        $<$<IN_LIST:${CMAKE_CXX_COMPILER_ID},XL>:>
```

Comment:
awesome how every compiler does this differently lol

Reply:
Truly thrilling.  There's a reason I just cut-and-paste from library to library, after having figured it out to a "good enough" state once

Reply:
PS: Also, `-Wall` (or whatever notation is used by a given compiler) is _not_ standardized and is likely gives a different set of warnings on every compiler...

Reply:
lol of course.